File: gateOne/Day Seven/phoneBook/phonebook.py
prompt = """
====================================================
SEMICOLON MOBILE PHONEBOOK APP (VERSION 1.0)
====================================================
1-> Add Contact
2-> Remove Contact
3-> Find Contact by Phone Number
4-> Find Contact by Firstname
5-> Find Contact by lastname
6-> Edit Contact

===================================================
""";

print(prompt)

firstname = []

lastname = [] 

phonenumber = []

user_Input = int(input("Enter any of the available Options to select"))

match(user_Input): 
	case 1 : add_Contact()

	case 2 : removeContact()

	case 3 : findContactByPhoneNumber()

	case 4 : findContactByFirstname()


def add_Contact():

	name = int(input("Enter your Firstname"))
  
	firstname.append(name)

	Lastnames = str(input("Enter your Lastname"))
  
	lastname.append(Lastnames)

	contact_number = str(input("Enter your Phone Number"))

	phonenumber.append(contact_number)


def remove_Contact():

	phone_number = str(input("Remove PhoneNumber"))
  
	for count in firstname: 
    
		if(contacts.get(count).equalsIgnoreCase(phoneNumber)):
        	
			phonenumber.remove(count)

	else:

		print("Contact does not exist")


def find_Contact_By_PhoneNumber():

	search = str(input("Search for contact"))

	for count in phonenumber:

		if(contacts.get(count).equalsIgnoreCase(phoneNumber)):
	
			contacts.get(count)
	
		print("Contact exists")
        
	else:

		print("Contacts cannot be found")


def findContactByFirstname():

	searchName = str(input("Search the Firstname of the Contact"))

	for count in firstname:

		if(searchName.equals(firstname.get(count))):

			firstname.indexOf(count)

		print(firstname + " exists")

	else: 
		("searchName" != (firstname.get(count)))

		firstname.indexOf(count)

		print("Firstname not found")


def findContactByLastname():

	lastnames = str(input("Search the Lastname of the Contact"))
    
	for count in lastname:

		if(lastName.equals(lastnames)):

			lastname.indexOf(lastName)
 
	print(lastName + " exists")

	if("lastName" != (lastnames)):

  		LastName.indexOf(lastName);
  
	print("LastName not found")
 



  

File: gateOne/Day Seven/phoneBook/phonebook.js
const prompt = require("prompt-sync")();

let firstname = [];

let lastname = [];

let phonenumber = [];

semicolonPhoneApp = phonebookMenu() 

function phonebookMenu() {

menuprompt = `
======================================================================
Welcome to Semicolon Phonebook App,Enter any of the Options to perform.

1-> Add Contact
2-> Remove Contact
3-> Find Contact by Phone Number
4-> Find Contact by Firstname
5-> Find Contact by lastname
6-> Edit Contact
7-> Back

=========================================
`;

console.log(menuprompt);

userInput = prompt("Select any operations to perform ");

switch (userInput){
    case "1": 
	    addContact();
    break;

    case "2": 
	    removeContact();
    break;

    case "3": 
	    findContactByPhoneNumber();
    break;

    case "4": 
	    findContactByFirstname();
    break;

    case "5": 
	    findContactByLastname();
    break;

    case "6": 
	    editContact();
    break;

    case "7": 
	    phonebookMenu();
    break;

     }
    
   }

function addContact()  {

name = prompt("Enter your Firstname ");

firstname.push(name);

names = prompt("Enter your Lastname ");
  
lastname.push(names);

number = prompt("Enter your Phone Number ");
  
phonenumber.push(number);

addAnotherContact();

}

function addAnotherContact()  {

newContact = prompt("Do you still want to add another Contact(Yes/No) ");
     
if(newContact === ("Yes")) {
      
 addContact();
 }

else {
 console.log("Contacts added Successfully>>>>>>>>>>>>>");
 phonebookMenu();
  }

}

function removeContact()  {

contactNumber = prompt("Remove Contact PhoneNumber ");

for(let count = 0; count < phonenumber.length; count++) {

if(contactNumber === (phonenumber.at(count)))  {
  phonenumber.pop(count);
  firstname.pop(count);
  lastname.pop(count);

 console.log(contactNumber.at(count));
   console.log(firstname);
    console.log(lastname);

console.log("Contact PhoneNumber deleted Successfully >>>>>>>>>>>");

console.log("Contact FirstName deleted Successfully >>>>>>>>>>>");

console.log("Contact LastName deleted Successsfully >>>>>>>>>>");
 } 

else {
    console.log("Contact does not exist");  
     }
 }

phonebookMenu();
    
}


function findContactByPhoneNumber()  {

searchNumber = prompt("Search for Phone Number");

   for(let count = 0; count < phonenumber.length; count++)  {

	if(searchNumber === (phonenumber.at(count)))  {
   	phonenumber.indexOf(count);
  	console.log(phonenumber + " exists");
   }

      if(searchNumber != (phonenumber.at(count)))  {
   	phonenumber.indexOf(count);
  	console.log("PhoneNumber not found");
    }
  }

phonebookMenu();
}

function findContactByFirstname() {

   searchName = ("Search the Firstname of the Contact");
           
 	for(let count = 0; count < firstname.length; count++) {
     
  	if(searchName === (firstname.at(count)))  {
   	firstname.indexOf(count);
  	console.log(firstname + " exists");
    }
 
       else if(searchName != (firstname.at(count)))  {
  	firstname.indexOf(count);
 	 console.log("Firstname not found");
   }
  }  
   
 } 
  
function findContactByLastname() {

lastNameSearch = prompt("Search the Lastname of the Contact");
    
for(let count = 0; count < lastname.length; count++) {
   
  if(lastNameSearch.equals(lastname.get(count))) {
   lastname.indexOf(count);
 console.log(lastname + " exists");
  }

 else if (!"lastNameSearch".equals(lastname.get(count))) {
  lastname.indexOf(count);
  console.log("Lastname not found");
  }
}

}

function editContact()  {

editContact = prompt("Enter the Contact you want to Edit");
    }

